### PR TITLE
create LxmertModelIntegrationTest Pytorch

### PR DESCRIPTION
**Pull request by sadakmed**
# What does this PR do?

this pr fix issue #9951 as it implements an integration test for  LXMERT

@LysandreJik

### CODE REVIEW

**Comment by LysandreJik** [comment]
Hello! It seems this is simply passing the test?

**Comment by sadakmed** [comment]
> Hello! It seems this is simply passing the test?

I made it to claim the issue and work on it.

**Comment by sadakmed** [comment]
@LysandreJik  Lxmert requires `visual_feats` and `visual_pos` could I change the `model.config.visual_feats_dim` to a smaller value like 5 or 10.

**Comment by sadakmed** [comment]
@LysandreJik  in this I used `np.random.seed` to fix the `visual_feats` and `visual_pos`,  otherwise we could load lxmertmodel by lxmertconfig while changing `visual_feat_dim` to something manageable. 

What do you suggest?

File: tests/test_modeling_lxmert.py [hunk 1]
```diff
@@ -726,3 +727,24 @@ def test_retain_grad_hidden_states_attentions(self):
...
         self.assertIsNotNone(attentions_vision.grad)
+
+@require_torch
+class LxmertModelIntegrationTest(unittest.TestCase):
+    @slow
+    def test_inference_no_head_absolute_embedding(self):
+        model = LxmertModel.from_pretrained(LXMERT_PRETRAINED_MODEL_ARCHIVE_LIST[0])
+        input_ids = torch.tensor([[101, 345, 232, 328, 740, 140, 1695, 69, 6078, 1588, 102]])
+        num_visual_features = 10
+        _, visual_feats = np.random.seed(0), np.random.rand(1,num_visual_features,LxmertModel.config.visual_feat_dim)
```
Comment (sadakmed) [inline tests/test_modeling_lxmert.py:738]: it looks ugly!!?

**Comment by sadakmed** [comment]
@LysandreJik  is there something I can help with here?

File: tests/test_modeling_lxmert.py [hunk 2]
```diff
@@ -726,3 +727,24 @@ def test_retain_grad_hidden_states_attentions(self):
...
+
+
+@require_torch
+class LxmertModelIntegrationTest(unittest.TestCase):
+    @slow
+    def test_inference_no_head_absolute_embedding(self):
+        model = LxmertModel.from_pretrained(LXMERT_PRETRAINED_MODEL_ARCHIVE_LIST[0])
+        input_ids = torch.tensor([[101, 345, 232, 328, 740, 140, 1695, 69, 6078, 1588, 102]])
+        num_visual_features = 10
+        _, visual_feats = np.random.seed(0), np.random.rand(1, num_visual_features, LxmertModel.config.visual_feat_dim)
```
Comment (LysandreJik) [inline tests/test_modeling_lxmert.py:739]: This should not be able to work, the `LxmertModel` has no attribute config. You probably meant `model` instead!

File: tests/test_modeling_lxmert.py [hunk 3]
```diff
@@ -726,3 +727,24 @@ def test_retain_grad_hidden_states_attentions(self):
...
+        _, visual_pos = np.random.seed(0), np.random.rand(1, num_visual_features, 4)
```
Comment (LysandreJik) [inline tests/test_modeling_lxmert.py:740]: I'm not super fond of setting the seed here as I feel it could have side effects across the tests.

**Review by LysandreJik — COMMENTED** [review]
Hi! Thanks for your work on this implementation! Indeed, it's probably one of the toughest ones. Could you try to replace the `np.random.rand` by the `ids_tensor` that are imported for other tests in this file? You can set the shape and the vocab size to ensure it's within the bounds of what you're looking for.

The same should apply to the TensorFlow test.

Thanks!

File: tests/test_modeling_lxmert.py (hunk 2 above)
Comment (sadakmed) [reply tests/test_modeling_lxmert.py:739]: yes, u r right!

File: tests/test_modeling_lxmert.py (hunk 3 above)
Comment (sadakmed) [reply tests/test_modeling_lxmert.py:740]: I tried a lot with this, and this way doesn't affect the randomness of others. 
besides, there's only one use of NumPy random in this test, so ...

**Comment by sadakmed** [comment]
> Could you try to replace the `np.random.rand` by the `ids_tensor`
well the `visual_feats` is torch.float will the `ids_tensor` returns int32.


I think another alternative, is to make the `model.config.visual_feats_num` smaller then we can have a fixed `visual_feats` ?


what do you think?

**Comment by sadakmed** [comment]
@LysandreJik

**Comment by sadakmed** [comment]
@LysandreJik  I did raise an issue to have a context manager that will fix a seed #10143, do you think it will be usefull here?

**Comment by sadakmed** [comment]
open

**Comment by LysandreJik** [comment]
Hello @sadakmed, sorry for taking a while to merge this. Do you mind rebasing on `master`, and running `make fixup` at the root of your clone? There's an issue with the code quality.

Will merge this right after.

**Comment by LysandreJik** [comment]
Thanks a lot for your contribution @sadakmed!
